Dockerfile

```diff
@@ -2,7 +2,7 @@
 # Read the docs/topics/development/docker.md file for more information about this Dockerfile.
 ####################################################################################################
 
-FROM python:3.11-slim-bookworm AS olympia
+FROM python:3.12-slim-bookworm AS olympia
```

Comment:
😄 literally no changes?

Reply:
Yeah. Looking at dependency updates the past couple years 99% of them added 3.12 to their compatibility matrix without any code changes.

Biggest breaking change is the removal of `distutils` by default but `setuptools` provides it for backwards-compatibility.